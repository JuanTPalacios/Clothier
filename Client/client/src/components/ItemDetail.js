import React from 'react'
import SearchResults from './SearchResults'
import { connect } from 'react-redux';
import { useState } from 'react';
import apiService from '../apiServices';
import '../styles/app.css';

function ItemDetail({searchVal, selectedItem, user}) {
  const [acquired, setAcquired] = useState(user.ADQs.map((item) => item.itemPrimaryKey))

  async function acquireItem () {
    await apiService.ADQ(user.primaryKey, selectedItem.primaryKey);
    setAcquired([...acquired, selectedItem.primaryKey]);
  }

  return (
    <div>
      {searchVal && 
      <div className="search">
      <SearchResults></SearchResults>
      </div>
      }
      Item Detail
      {selectedItem.title}
      <img src={selectedItem.image} alt="n/a" />
      {acquired.includes(selectedItem.primaryKey)
      ? <h6>Already in MyCloset</h6>
      : <button onClick={() => acquireItem()}>Add to MyCloset</button>
      }
    </div>
  )
}

const mapStateToProps = ({store}) => {
  return {
    user: store.user,
    searchVal: store.searchVal,
    selectedItem: store.selectedItem,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ItemDetail);


