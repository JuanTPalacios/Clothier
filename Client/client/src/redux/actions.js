const actions = {
  getItems : (items) => ({
    type: 'getItems',
    payload: items
  }),
};

export default actions;