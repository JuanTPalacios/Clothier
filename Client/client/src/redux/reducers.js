import { combineReducers } from 'redux';

const initialState = {
  items: [] ,
  setAuthenticated: false
}

const items = (state = initialState, action) => {
  switch (action.type) {
    case 'getItems':
    return {items: [...state.items, ...action.payload]};
  default:
    return state;
  }
};

// Combining both reducers
const reducers = combineReducers({
  items,
});

export default reducers;