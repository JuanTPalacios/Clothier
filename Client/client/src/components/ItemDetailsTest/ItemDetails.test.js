import {render, screen} from '@testing-library/react';
import { createStore } from 'redux';
import ReactRouter from 'react-router'
import { Provider } from 'react-redux';
import reducers from '../../redux/reducers';
import ItemDetail from './ItemDetail';

describe('Item Details component', () => {
  test('should match the snapshot', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ itemId: 226 });
    let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__());
    const { container } = render(<Provider store={store}><ItemDetail /></Provider>);
    expect(container.firstChild).toMatchSnapshot();
  })

  test('It should render the corrrect component and text', () => {

    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ itemId: 226 })
    let store = createStore(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ &&
        window.__REDUX_DEVTOOLS_EXTENSION__());
    render(<Provider store={store}><ItemDetail/></Provider>);
    screen.getByText(/Add to MyCloset/)
    screen.getByText(/Similar clothes in your Closet:/)
    screen.getByText(/Other clothes in this Category:/)
    screen.getByText(/Similar clothes in your Closet:/)
    screen.getByRole('button', {name: 'Add to MyCloset'})
  })
  


})
  