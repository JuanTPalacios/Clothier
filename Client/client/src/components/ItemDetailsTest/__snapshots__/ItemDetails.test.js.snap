// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Details component should match the snapshot 1`] = `
<div
  class="content item"
>
  
  <div
    class="box item-detail columns"
  >
    <div
      class="column card"
    >
      <img
        alt="n/a"
      />
    </div>
    <div
      class="column card"
    >
      <h1
        class="title is-3"
      />
      <h1
        class="title is-4"
      />
      <h1
        class="title is-5"
      />
      <button
        class="button is-rounded is-success m-2"
      >
        Add to MyCloset
      </button>
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="button is-rounded is-success m-2"
        >
          Buy
        </button>
      </a>
    </div>
  </div>
  <div
    class="box columns item-related"
  >
    <div
      class="column card"
    >
      <h1
        class="title is-5"
      >
        Similar clothes in your Closet:
      </h1>
      <div
        class="search-follow"
      />
    </div>
    <div
      class="column card"
    >
      <h1
        class="title is-5"
      >
        Other clothes in this Category:
      </h1>
      <div
        class="search-follow"
      />
    </div>
  </div>
</div>
`;
